src/app/store/account.slice.ts

```diff
@@ -64,37 +77,51 @@ export const getAccounts = createSelector(
  */
 export function* accountsSaga() {
   yield all([
-    takeLatest(fetchAccount.type, fetchAccountWorker),
+    takeLatest(fetchAccounts.type, fetchAccountsWorker),
     takeLatest(removeAccount.type, removeAccountWorker)
   ]);
 }
 
-export function* fetchAccountWorker({
-  payload: wallet
-}: PayloadAction<SerializedWallet & { persistent: boolean }>) {
-  const address: TAddress = yield call(ipcBridgeRenderer.crypto.invoke, {
-    type: CryptoRequestType.GET_ADDRESS,
-    wallet
-  });
-
-  const uuid = generateDeterministicAddressUUID(address);
-
-  if (wallet.persistent) {
-    yield call(ipcBridgeRenderer.db.invoke, {
-      type: DBRequestType.SAVE_ACCOUNT_SECRETS,
-      wallet
-    });
+export function* fetchAccountsWorker({
+  payload: wallets
+}: PayloadAction<(SerializedWallet & { persistent: boolean })[]>) {
+  const accounts: IAccount[] = yield select(getAccounts);
+
+  try {
+    for (let i = 0; i < wallets.length; i++) {
+      const wallet = wallets[i];
+      const address: TAddress = yield call(ipcBridgeRenderer.crypto.invoke, {
+        type: CryptoRequestType.GET_ADDRESS,
+        wallet
+      });
+
+      const uuid = generateDeterministicAddressUUID(address);
+
+      if (accounts.find((a) => a.uuid === uuid)) {
+        continue;
```

Comment:
Should we replace the previous entry, rather than ignoring it? For example if a user has a private key, but decides to use a mnemonic phrase later on?

Reply:
Good question, I don't know entirely what would be the logical way to handle it